### ISSUE TITLE
Allow `data_source="X_y"` for `CrossValidationReport`

### ISSUE THREAD

**Issue by glemaitre** [probabl-ai/skore#1500]
### What would you like to say?

For the moment, we don't support the case `data_source="X_y"` for `CrossValidationReport`. Actually, I think that there potential use cases to make predict all the estimators from the cross-validation on the same set to know the variance or misclassification samples.

So I think that we could go ahead and implement the feature and at least support the metric for the moment. When it comes to plotting, we need to have specific representation for this case most probably.

**Comment by MarieSacksick** [comment]
> I think that there potential use cases to make predict all the estimators from the cross-validation on the same set

It's not clear to me why the user would want to use the estimators in each fold for cross validation, could you elaborate please? I find a bit odd, since these estimators are just done for statistical reasons, and in no way are meant to be further used.

**Comment by glemaitre** [comment]
> It's not clear to me why the user would want to use the estimators in each fold for cross validation, could you elaborate please? 

Here, you have to see this external dataset as an extra test set where I want to see the variations of each estimator on the same set. Cross-validation will not allow me to see variation on the same sample (apart of using `ShuffleSplit` with a lot of split). It potentially help you at debugging and understanding what happen with one specific model and most probably the relationship between a model and the training data.

> I find a bit odd, since these estimators are just done for statistical reasons, and in no way are meant to be further used.

Here, it is not really what I want to study: I'm not really going to use any of those models in production later.

However, sometimes it is actually a use case: the `CalibratedClassifierCV` is doing such an ensemble and we discussed in the past that `GridSearchCV` could use the ensemble of the model with the best parameters instead of doing a `refit` strategy.
